### ISSUE TITLE
Local server (Ollama) model discovery does not work? (since 5.1-5.2)

### ISSUE THREAD

**Issue by Gaspadlo** [SystemSculpt/obsidian-systemsculpt-ai#29]
Hello,

I am running local Ollama server with multiple models (Win Obsidian + Win Ollama).
On version 5.0 selecting available models worked, but on latest version (5.2), the plugin reports no models...

Manually reverting main.js; style.css; manifest.json back from 5.0 release, fixed the models discovery.

http://127.0.0.1:11434/api/tags consistently returns

```
{
    "models": [
        {
            "name": "llama3.1:latest",
            "model": "llama3.1:latest",
            "modified_at": "2024-07-25T00:01:35.9451235+02:00",
            "size": 4661226402,
            "digest": "62757c860e01d552d4e46b09c6b8d5396ef9015210105427e05a8b27d7727ed2",
            "details": {
                "parent_model": "",
                "format": "gguf",
                "family": "llama",
                "families": [
                    "llama"
                ],
                "parameter_size": "8.0B",
                "quantization_level": "Q4_0"
            }
        },
        {
            "name": "gemma2:latest",
            "model": "gemma2:latest",
            "modified_at": "2024-07-24T22:54:23.4535771+02:00",
            "size": 5443152417,
            "digest": "ff02c3702f322b9e075e9568332d96c0a7028002f1a5a056e0a6784320a4db0b",
            "details": {
                "parent_model": "",
                "format": "gguf",
                "family": "gemma2",
                "families": [
                    "gemma2"
                ],
                "parameter_size": "9.2B",
                "quantization_level": "Q4_0"
            }
        }
    ]
}
```

**Comment by SystemSculpt** [comment]
Hi! Looks like you're using ollama - I recently removed the api/tags endpoint check because Ollama finally supports the typical /v1/models endpoint everyone else uses. Have you updated your Ollama recently?

**Comment by SystemSculpt** [comment]
Though regardless I should still have /api/tags as a fallback, so I will add this back in anyway, but I would like to see if this is the root of the problem and nothing else more complicated so I would love if you confirmed :D

**Comment by Gaspadlo** [comment]
I guess Ollama for Windows (Preview) is not so up-to-date at the moment...
I could run it in WSL2, but I wanted to avoid that :)

**Comment by SystemSculpt** [comment]
Totally understand ya, I'll get the api/tags endpoint going again in the next update :) i'll keep this open until it's done

**Comment by Andrej4711** [comment]
Same issue with LM Studio 0.2.28.

**Comment by SystemSculpt** [comment]
Interesting. I'll do some thorough testing and get a fix out somewhere in the next 12-24 hours

**Comment by niesink** [comment]
Might be the same underlying bug as this? https://github.com/SystemSculpt/obsidian-systemsculpt-ai/issues/30#issuecomment-2252480240

**Comment by SystemSculpt** [comment]
Should be fixed now with latest 0.5.3 update :) if not, let me know by re-opening/creating a new issue. Thanks and enjoy!
